mongorestore/mongorestore_test.go

```diff
@@ -107,12 +117,86 @@ func TestMongorestore(t *testing.T) {
 			restore.NSOptions.DB = "db1"
 			restore.InputReader = bsonFile
 			restore.TargetDirectory = "-"
+
+			result := restore.Restore()
+			So(result.Err, ShouldBeNil)
+			count, err := c1.CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 100)
+
+			restore.NSOptions.Collection = ""
+			restore.NSOptions.DB = ""
+			restore.InputReader = nil
+			restore.TargetDirectory = ""
+			c1.Drop(nil)
```

Comment:
These options should be reset after each Convey block (otherwise we get conflicting options later on). Dropping the collections also lets us run tests repeatedly without manually dropping them on the command line each time.

Reply:
I don't think you should need to reset the options. Take a look at this page in the goconvey docs: https://github.com/smartystreets/goconvey/wiki/Execution-order

In general teardown logic such as dropping collections should be in the Reset function (https://github.com/smartystreets/goconvey/wiki/Reset) or, if you prefer, in a defer statement in the parent convey block.

Reply:
That's really interesting actually! I didn't realize that it ran like the Convey blocks like nodes in a tree :)

I'll just remove these lines then. The collections will persist in `db1` as before, but that doesn't affect successive runs (I was probably doing something wrong before).